tests/src/Kernel/UniqueEmailTest.php

```diff
@@ -0,0 +1,586 @@
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user fails validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'all' mode with an existing coupled
+   * user.
+   *
+   * @covers ::validate
+   */
+  public function testModeAllCoupled() {
+    $this->setUniqueEmailsConfig('all');
+
+    $existing_user = $this->createUser();
+
+    // Test validating a decoupled user.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->getAccountName());
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Decoupled user fails validation.');
+
+    // Test validating a coupled user.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->getAccountName());
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Coupled user fails validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'none' mode with no existing users.
+   *
+   * @covers ::validate
+   */
+  public function testModeNoneNone() {
+    $this->setUniqueEmailsConfig('none');
+
+    // Test validating a decoupled user.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser();
+    $violations = $user->validate();
+    $this->assertEmpty($violations, 'Decoupled user passes validation.');
+
+    // Test validating a coupled user.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser(FALSE);
+    $violations = $user->validate();
+    $this->assertEmpty($violations, 'Coupled user passes validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'none' mode with an existing decoupled
+   * user.
+   *
+   * @covers ::validate
+   */
+  public function testModeNoneDecoupled() {
+    $this->setUniqueEmailsConfig('none');
+
+    $existing_user = $this->createDecoupledUser();
+
+    // Test validating a decoupled user.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->email_prefix);
+    $violations = $user->validate();
+    $this->assertEmpty($violations, 'Decoupled user passes validation.');
+
+    // Test validating a coupled user.
+    // Expected result: No violations
+    $user = $this->createUnsavedUser(FALSE, $existing_user->email_prefix);
+    $violations = $user->validate();
+    $this->assertEmpty($violations, 'Coupled user passes validation.');
+  }
+
+  /**
+   * Test the unique email validator in 'none' mode with an existing coupled
+   * user.
+   *
+   * @covers ::validate
+   */
+  public function testModeNoneCoupled() {
+    $this->setUniqueEmailsConfig('none');
+
+    $existing_user = $this->createUser();
+
+    // Test validating a decoupled user.
+    // Expected result: Violations
+    $user = $this->createUnsavedUser(TRUE, $existing_user->getAccountName());
+    $violations = $user->validate();
+    $this->assertNotEmpty($violations, 'Decoupled user fails validation.');
```

Comment:
Shouldn't this be asserting empty? If the mode is none, then a decoupled user can have the same email address as a coupled user.